plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php

```diff
@@ -62,7 +62,13 @@ function articleViewCallback($hookName, $args) {
 		$galley =& $args[2];
 		$article =& $args[3];
 
-		if ($galley && $galley->getFileType() == 'text/html') {
+		if (!$galley) {
+			return false;
+		}
+
+		$submissionFile = $galley->getFile();
+		$filepath = Services::get('file')->getPath($submissionFile->getData('fileId'));
+		if (Services::get('file')->fs->getMimetype($filepath) === 'text/html') {
```

Comment:
Have you checked how good `getMimeType` is at detecting HTML documents? The underlying OS tools are pretty bad, and we had to add hints to the system to use the file extension in cases like this. Not sure if that's what's happening under the hood with Flysystem.

Reply:
Do you have some examples of where it went badly? I tried testing with a `test.html`, a `test.shtml` and a`test.xhtml` file. That gave me `text/html` in each case. However, when I removed `<html>` from the file I got `text/plain`.

If we have some example cases where we want to ensure the correct mimetype is returned, or alter the mimetype to account for badly formatted HTML, I could maybe build those into our test suite.

Reply:
It looks like this uses `detectMimeTypeFromPath` from the [mime-type-detection](https://github.com/thephpleague/mime-type-detection) library.

Reply:
`detectMimeTypeFromPath` will do fine -- the current `PKPString::mime_content_type` function is probably equivalent to `detectMimeType`, in that it uses the file contents (not just the extension) to detect. Detection by contents fails badly with HTML files, so we added some quirks to prioritize typing by file extension in certain cases. It woudl be wonderful to do away with all of this, and `PKPString::mime_content_type`, using the mime-type-detection library!